src/ios/CDVOrientation.m

```diff
@@ -80,7 +80,13 @@ -(void)screenOrientation:(CDVInvokedUrlCommand *)command
             }
             if (value != nil) {
                 _isLocked = true;
-                [[UIDevice currentDevice] setValue:value forKey:@"orientation"];
+                if (@available(iOS 16.0, *)) {
+                    #if __IPHONE_OS_VERSION_MAX_ALLOWED > __IPHONE_15_5 // Xcode 14 and iOS 16, or greater
+                        [self.viewController setNeedsUpdateOfSupportedInterfaceOrientations];
+                    #endif
```

Comment:
So I take it if the application was compiled using XCode 13, and this was ran on an iOS 16 device, this effectively becomes a no-op, as it will enter the `if (@available(iOS 16.0, *)) {` condition, but only XCode >= 14 will actually have code.

Is my understanding correct?

Reply:
```
#if __IPHONE_OS_VERSION_MAX_ALLOWED > __IPHONE_15_5
if (@available(iOS 16.0, *)) {
  [self.viewController setNeedsUpdateOfSupportedInterfaceOrientations];
} else
#endif
{
  [UINavigationController attemptRotationToDeviceOrientation];
}
```

Should be better.

Reply:
hello. what use case is it better for ? thanks

Reply:
This is only to make the CI pass.

The API that the OP has introduced in this PR is not available on older Xcode versions. The test fails to compile the code as it is referencing an unknown API to the Xcode version.

The above code will check the iOS version that correlates with an Xcode version. If the version of Xcode does not support the iOS version, the block of code that contains the new API is removed before compile.

Reply:
Thanks for the suggestion. As I stated before, Swift is not my specialty area. I used my best judgement to update what you suggested. Hopefully I got it right? 😅

https://github.com/apache/cordova-plugin-screen-orientation/pull/102/commits/8ab74f799f0db3087ea309264b48949f7814065f